kernel/src/main/scala/cats/kernel/instances/symbol.scala

```diff
@@ -0,0 +1,15 @@
+package cats.kernel
+package instances
+
+package object symbol extends SymbolInstances
+
+trait SymbolInstances {
+  implicit val catsKernelStdOrderForSymbol: Order[Symbol] = new SymbolOrder
+}
+
+class SymbolOrder extends Order[Symbol] {
+  override def eqv(x: Symbol, y: Symbol): Boolean =
+    x eq y
```

Comment:
Any reason not to use == here? I assume Symbol's equal method is correct and we don't need to worry here that eq is correct.